src/EKI.jl

```diff
@@ -134,6 +147,43 @@ function update_ensemble!(eki::EKIObj{FT}, g) where {FT}
 
     compute_error(eki)
 
+    # Check convergence
+    cov_new = cov(eki.u[end], dims=1)
+    cov_threshold = 0.01
+    if det(cov_new) < cov_threshold*det(cov_init)
+        @warn "Ensemble covariance after the last EKI stage decreased significantly. 
```

Comment:
No matter whether `cov_threshold` ends up being hard-coded or not, I would replace "significantly" with the actual quantitative information about how much the covariance has decreased in that iteration.

Reply:
I agree, specially after your previous comment and allowing a user based `con_threshold`.